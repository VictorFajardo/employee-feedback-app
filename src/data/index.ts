import { EmployeeInterface, ReviewInterface } from '../interfaces';

export const DefaultEmployeeFields: EmployeeInterface = {
  id: '',
  createdAt: '',
  firstName: '',
  lastName: '',
  jobTitle: '',
  email: '',
  password: '',
  confirmPassword: '',
  admin: false,
  reviews: [],
};

export const DefaultReviewFields: ReviewInterface = {
  id: '',
  createdAt: '',
  employeeName: '',
  employeeJobTitle: '',
  employeeEmail: '',
  reviewerName: '',
  reviewerJobTitle: '',
  reviewerEmail: '',
  content: '',
  completed: false,
};

export enum ModalType {
  VIEW,
  UPDATE,
  ADD,
  DELETE,
}
