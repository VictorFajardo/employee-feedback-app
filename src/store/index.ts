import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import employeesSlice from '../features/employees/employeesSlice';
import reviewsSlice from '../features/reviews/reviewsSlice';
import authenticationSlice from '../features/authentication/authenticationSlice';

export const store = configureStore({
  reducer: {
    employees: employeesSlice,
    reviews: reviewsSlice,
    authentication: authenticationSlice,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
