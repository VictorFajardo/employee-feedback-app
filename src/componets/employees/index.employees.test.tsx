import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { store } from '../../store';
import IndexEmployees from './index.employees';
import { ModalType } from '../../data';

it('expect render IndexEmployees component', () => {
  const mockProps = {
    modal: ModalType.VIEW,
    closeMethod: () => {},
    clickMethod: () => {},
  };
  expect(
    shallow(
      <Provider store={store}>
        <IndexEmployees {...mockProps} />
      </Provider>
    )
  ).toMatchSnapshot();
});
