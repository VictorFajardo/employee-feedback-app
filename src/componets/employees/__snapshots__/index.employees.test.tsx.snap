// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect render Index component 1`] = `ShallowWrapper {}`;

exports[`expect render IndexEmployees component 1`] = `ShallowWrapper {}`;
